File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> st;
        int maxSize = 0;
        int j = 0;
        int start=0;
        for(int start = 0; start < s.length(); start++) {
    set<char> st;

    for(int i = start; i < s.length(); i++) {
        if(st.count(s[i]))
            break;

        st.insert(s[i]);
    }

    maxSize = max(maxSize, (int)st.size());
}
       maxSize = max(maxSize, (int)st.size());
        return maxSize;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxSize = 0
        for start in range(len(s)):
            st = set()
            for i in range(start, len(s)):
                if s[i] in st:
                    break
                st.add(s[i])
            maxSize = max(maxSize, len(st))
        return maxSize